.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+      fail-fast: false
+      matrix:
+        repository: ${{ fromJSON(inputs.repositories) }}
+        runner: ${{ fromJSON(inputs.runners) }}
+        command: ${{ fromJSON(inputs.commands) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: ${{ github.repository }}
+      - if: runner.os == 'Windows'
+        run: choco install yq
+      - id: workflow
+        run: |
+          echo "config<<EOF" >> $GITHUB_OUTPUT
+          yq -o json . .github/config/regression-tests.yml >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+        working-directory: ${{ github.repository }}
+      - id: config
+        env:
+          REPOSITORY: ${{ matrix.repository }}
+          RUNNER: ${{ matrix.runner }}
+          COMMAND: ${{ matrix.command }}
+          CONFIG: ${{ toJSON(fromJSON(steps.workflow.outputs.config)) }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const config = JSON.parse(process.env.CONFIG);
+            const repositoryConfig = config.repositories[process.env.REPOSITORY];
+            const runnerConfig = config.runners[process.env.RUNNER];
+            const commandConfig = config.commands[process.env.COMMAND];
+            if (!repositoryConfig) {
+              core.setFailed(`Unsupported repository: ${process.env.REPOSITORY}`);
+            }
+            if (!runnerConfig) {
+              core.setFailed(`Unsupported runner: ${process.env.RUNNER}`);
+            }
+            if (!commandConfig) {
+              core.setFailed(`Unsupported command: ${process.env.COMMAND}`);
+            }
+            core.setOutput('repository', JSON.stringify(repositoryConfig));
+            core.setOutput('runner', JSON.stringify(runnerConfig));
+            core.setOutput('command', JSON.stringify(commandConfig));
+            console.log({
+              repository: repositoryConfig,
+              runner: runnerConfig,
+              command: commandConfig,
+            });
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+      - name: Download Hardhat
+        if: (startsWith(matrix.command, 'hardhat') && inputs.hardhat-ref != 'next' && !startsWith(inputs.hardhat-ref, '3'))
+        uses: actions/download-artifact@v4
+        with:
+          name: hardhat
+      - name: Fix Hardhat on Windows
+        if: (startsWith(matrix.command, 'hardhat') && inputs.hardhat-ref != 'next' && !startsWith(inputs.hardhat-ref, '3')) && matrix.runner == 'windows-latest'
+        run: |
+          npm init -y
+          npm install esbuild
+          rm package.json package-lock.json
+          tar -xzf hardhat-*.tgz
+          rm -rf package/node_modules/esbuild package/node_modules/@esbuild
+          mv node_modules/esbuild package/node_modules
+          mv node_modules/@esbuild package/node_modules
+          tar -czf hardhat-*.tgz package
+          rm -rf package
```

Comment:
The esbuild binary inside the hardhat tarball cannot be executed on windows. That's why we replace it with a system specific one.